mitodl/ocw-studio
# Add markdown support to CKEditor

#### Pre-Flight checklist


#### What are the relevant tickets?

part of #39 

#### What's this PR do?

This adds support for Markdown to our CKEditor component. How?

- a markdown library (`lib/markdown.ts`) exporting two functions:
    - `md2html`: supports translating Markdown to HTML. Implemented with [showdown](https://github.com/showdownjs/showdown/)
    - `html2md`: translates HTML to Markdown. This is implemented with the customized turndown from the `@ckeditor/ckeditor5-markdown-gfm`
- a `DataProcessor` plugin for CKEditor which subclasses `GFMDataProcessor` from `@ckeditor/ckeditor5-markdown-gfm`. This basically provides for transforming the input and output of the editor from / to Markdown.

I also added a fix for some strange default behavior in Turndown (using ` *` to denote lists instead of `- `).

#### How should this be manually tested?

test out the editor on the `/markdown-editor` page. Fiddle with the stuff in the editor and check out the output below, and just generally make sure nothing too weird is going on. Should be pretty straightforward I think!

#### Screenshots (if appropriate)

![Screenshot from 2021-02-25 16-16-31](https://user-images.githubusercontent.com/6207644/109218896-000ade00-7785-11eb-9108-791b1dcc7737.png)
